grpc-bridge/src/main/java/org/jboss/resteasy/grpc/servlet/MockServletInputStream.java

```diff
@@ -0,0 +1,38 @@
+package org.jboss.resteasy.grpc.servlet;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import jakarta.servlet.ReadListener;
+import jakarta.servlet.ServletInputStream;
+
+public class MockServletInputStream extends ServletInputStream {
+   private InputStream is;
+
+   public MockServletInputStream(final InputStream is) {
+      this.is = is;
+   }
+
+   @Override
+   public boolean isFinished() {
+      try {
+         return is.available() > 0;
```

Comment:
This doesn't look right to me. There is no guarantee or contract that `InputStream.available()` must return a value other than 0 even if there is data available to be read.

Reply:
A MockServletInputStream is always created as follows:

      ByteArrayInputStream bais = new ByteArrayInputStream(actualParam.toByteArray());
      MockServletInputStream msis = new MockServletInputStream(bais);

and available() in ByteArrayInputStream is

    public synchronized int available() {
        return count - pos;
    }